sdk/rust/src/processor/handler.rs

```diff
@@ -389,7 +389,7 @@ impl TransactionContext {
         let x: &[u8] = &serialized;
 
         let mut future = self.sender.send(
-            Message_MessageType::TP_RECEIPT_ADD_DATA_REQUEST,
+            Message_MessageType::TP_EVENT_ADD_REQUEST,
```

Comment:
This fix should go in its own commit (or in its own PR since multiple of your pr depend on this fix)